src/App.tsx

```diff
@@ -15,48 +17,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_BY = {
+  ALPHA: 'alpha',
+  LENGTH: 'length',
+};
+
+type SortingParams = {
+  sortBy: string,
+  isReverse: boolean
+};
+
+function getVisibleGoods(
+  goods:string[],
+  {
+    sortBy,
+    isReverse,
+  }: SortingParams,
+): string[] {
+  const result = [...goods];
+
+  if (sortBy) {
+    result.sort((a, b) => {
+      switch (sortBy) {
+        case SORT_BY.ALPHA:
+          return a.localeCompare(b);
+
+        case SORT_BY.LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    result.reverse();
+  }
+
+  return result;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
```

Comment:
let's add default value for SORT_BY
```suggestion
  const [sortBy, setSortBy] = useState(SORT_BY.default);
```